### ISSUE TITLE
Brave Helper (0.9.6) constant crash with EXC_BAD_INSTRUCTION

### ISSUE THREAD

**Issue by fiery-** [brave/browser-laptop#1772]
```
Process:               Brave Helper [1558]
Path:                  /Applications/Brave.app/Contents/Frameworks/Brave Helper.app/Contents/MacOS/Brave Helper
Identifier:            com.electron.brave.helper
Version:               0
Code Type:             X86-64 (Native)
Parent Process:        Brave [1555]
Responsible:           Brave Helper [1558]
User ID:               501

Date/Time:             2016-05-15 21:00:41.553 +0300
OS Version:            Mac OS X 10.11.4 (15E65)
Report Version:        11

Time Awake Since Boot: 20000 seconds
Time Since Wake:       2600 seconds

System Integrity Protection: enabled

Crashed Thread:        0  CrRendererMain  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_INSTRUCTION (SIGILL)
Exception Codes:       0x0000000000000001, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Thread 0 Crashed:: CrRendererMain  Dispatch queue: com.apple.main-thread
0   com.brave.brave.framework     	0x000000010fed773f 0x10dad0000 + 37779263
1   libnode.dylib                 	0x00000001127332dd v8::internal::Isolate::FireBeforeCallEnteredCallback() + 45
2   libnode.dylib                 	0x0000000112391375 v8::Function::Call(v8::Local<v8::Context>, v8::Local<v8::Value>, int, v8::Local<v8::Value>*) + 213
3   libnode.dylib                 	0x0000000112030483 node::MakeCallback(node::Environment*, v8::Local<v8::Value>, v8::Local<v8::Function>, int, v8::Local<v8::Value>*) + 707
4   libnode.dylib                 	0x0000000112030808 node::MakeCallback(node::Environment*, v8::Local<v8::Object>, v8::Local<v8::String>, int, v8::Local<v8::Value>*) + 72
5   libnode.dylib                 	0x0000000112030866 node::MakeCallback(node::Environment*, v8::Local<v8::Object>, char const*, int, v8::Local<v8::Value>*) + 70
6   libnode.dylib                 	0x00000001120308e0 node::MakeCallback(v8::Isolate*, v8::Local<v8::Object>, char const*, int, v8::Local<v8::Value>*) + 96
7   com.brave.brave.framework     	0x000000010db9f708 mate::internal::CallEmitWithArgs(v8::Isolate*, v8::Local<v8::Object>, std::__1::vector<v8::Local<v8::Value>, std::__1::allocator<v8::Local<v8::Value> > >*) + 104
8   com.brave.brave.framework     	0x000000010dbc5205 0x10dad0000 + 1004037
9   com.brave.brave.framework     	0x000000010dbc4d47 atom::AtomRendererClient::WillReleaseScriptContext(v8::Local<v8::Context>) + 55
10  com.brave.brave.framework     	0x000000010dbc4fee 0x10dad0000 + 1003502
11  com.brave.brave.framework     	0x000000010e942ce9 0x10dad0000 + 15150313
12  com.brave.brave.framework     	0x000000010fedf7bb 0x10dad0000 + 37812155
13  com.brave.brave.framework     	0x000000010fee1b93 0x10dad0000 + 37821331
14  com.brave.brave.framework     	0x000000010fea743e 0x10dad0000 + 37581886
15  com.brave.brave.framework     	0x000000011044cddc 0x10dad0000 + 43503068
16  com.brave.brave.framework     	0x00000001105411ff 0x10dad0000 + 44503551
17  com.brave.brave.framework     	0x000000010df92283 0x10dad0000 + 4989571
18  com.brave.brave.framework     	0x000000010e9705fd 0x10dad0000 + 15336957
19  com.brave.brave.framework     	0x000000010e96501f 0x10dad0000 + 15290399
20  com.brave.brave.framework     	0x000000010dcb219b 0x10dad0000 + 1974683
21  com.brave.brave.framework     	0x000000010f7f37c3 0x10dad0000 + 30554051
22  com.brave.brave.framework     	0x000000010f7f2579 0x10dad0000 + 30549369
23  com.brave.brave.framework     	0x000000010f7f43f3 0x10dad0000 + 30557171
24  com.brave.brave.framework     	0x000000010dcb219b 0x10dad0000 + 1974683
25  com.brave.brave.framework     	0x000000010dcd0973 0x10dad0000 + 2099571
26  com.brave.brave.framework     	0x000000010dcd0c4c 0x10dad0000 + 2100300
27  com.brave.brave.framework     	0x000000010dcd0e3b 0x10dad0000 + 2100795
28  com.brave.brave.framework     	0x000000010dca7a41 0x10dad0000 + 1931841
29  com.brave.brave.framework     	0x000000010dcc891a 0x10dad0000 + 2066714
30  com.brave.brave.framework     	0x000000010dca7444 0x10dad0000 + 1930308
31  com.apple.CoreFoundation      	0x00007fff8e886881 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
32  com.apple.CoreFoundation      	0x00007fff8e865fbc __CFRunLoopDoSources0 + 556
33  com.apple.CoreFoundation      	0x00007fff8e8654df __CFRunLoopRun + 927
34  com.apple.CoreFoundation      	0x00007fff8e864ed8 CFRunLoopRunSpecific + 296
35  com.apple.Foundation          	0x00007fff95b5fdd9 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 270
36  com.brave.brave.framework     	0x000000010dca80ae 0x10dad0000 + 1933486
37  com.brave.brave.framework     	0x000000010dca7894 0x10dad0000 + 1931412
38  com.brave.brave.framework     	0x000000010dce3743 0x10dad0000 + 2176835
39  com.brave.brave.framework     	0x000000010dcd016d 0x10dad0000 + 2097517
40  com.brave.brave.framework     	0x000000010e977654 0x10dad0000 + 15365716
41  com.brave.brave.framework     	0x000000010e376f9c 0x10dad0000 + 9072540
42  com.brave.brave.framework     	0x000000010e3764b6 0x10dad0000 + 9069750
43  com.brave.brave.framework     	0x000000010dad2353 AtomMain + 83
44  com.electron.brave.helper     	0x000000010dac8e56 main + 38
45  libdyld.dylib                 	0x00007fff9a80e5ad start + 1

Thread 1:: Dispatch queue: com.apple.libdispatch-manager
0   libsystem_kernel.dylib        	0x00007fff99f49efa kevent_qos + 10
1   libdispatch.dylib             	0x00007fff96559165 _dispatch_mgr_invoke + 216
2   libdispatch.dylib             	0x00007fff96558dcd _dispatch_mgr_thread + 52

Thread 2:: Chrome_ChildIOThread
0   libsystem_kernel.dylib        	0x00007fff99f49eca kevent + 10
1   com.brave.brave.framework     	0x000000010ea93117 0x10dad0000 + 16527639
2   com.brave.brave.framework     	0x000000010ea909db 0x10dad0000 + 16517595
3   com.brave.brave.framework     	0x000000010dca6d61 0x10dad0000 + 1928545
4   com.brave.brave.framework     	0x000000010dce3743 0x10dad0000 + 2176835
5   com.brave.brave.framework     	0x000000010dcd016d 0x10dad0000 + 2097517
6   com.brave.brave.framework     	0x000000010dd02aa8 0x10dad0000 + 2304680
7   com.brave.brave.framework     	0x000000010dcfec47 0x10dad0000 + 2288711
8   libsystem_pthread.dylib       	0x00007fff9734199d _pthread_body + 131
9   libsystem_pthread.dylib       	0x00007fff9734191a _pthread_start + 168
10  libsystem_pthread.dylib       	0x00007fff9733f351 thread_start + 13

Thread 3:: CompositorTileWorker1/17435
0   libsystem_kernel.dylib        	0x00007fff99f48db6 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0x00007fff97342728 _pthread_cond_wait + 767
2   com.brave.brave.framework     	0x000000010e92a169 0x10dad0000 + 15049065
3   com.brave.brave.framework     	0x000000010dd0206d 0x10dad0000 + 2302061
4   com.brave.brave.framework     	0x000000010dcfec47 0x10dad0000 + 2288711
5   libsystem_pthread.dylib       	0x00007fff9734199d _pthread_body + 131
6   libsystem_pthread.dylib       	0x00007fff9734191a _pthread_start + 168
7   libsystem_pthread.dylib       	0x00007fff9733f351 thread_start + 13

Thread 4:: CompositorTileWorker2/13899
0   libsystem_kernel.dylib        	0x00007fff99f48db6 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0x00007fff97342728 _pthread_cond_wait + 767
2   com.brave.brave.framework     	0x000000010e92a169 0x10dad0000 + 15049065
3   com.brave.brave.framework     	0x000000010dd0206d 0x10dad0000 + 2302061
4   com.brave.brave.framework     	0x000000010dcfec47 0x10dad0000 + 2288711
5   libsystem_pthread.dylib       	0x00007fff9734199d _pthread_body + 131
6   libsystem_pthread.dylib       	0x00007fff9734191a _pthread_start + 168
7   libsystem_pthread.dylib       	0x00007fff9733f351 thread_start + 13

Thread 5:: CompositorTileWorker3/18195
0   libsystem_kernel.dylib        	0x00007fff99f48db6 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0x00007fff97342728 _pthread_cond_wait + 767
2   com.brave.brave.framework     	0x000000010e92a169 0x10dad0000 + 15049065
3   com.brave.brave.framework     	0x000000010dd0206d 0x10dad0000 + 2302061
4   com.brave.brave.framework     	0x000000010dcfec47 0x10dad0000 + 2288711
5   libsystem_pthread.dylib       	0x00007fff9734199d _pthread_body + 131
6   libsystem_pthread.dylib       	0x00007fff9734191a _pthread_start + 168
7   libsystem_pthread.dylib       	0x00007fff9733f351 thread_start + 13

Thread 6:: Compositor
0   libsystem_kernel.dylib        	0x00007fff99f42f72 mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x00007fff99f423b3 mach_msg + 55
2   com.apple.CoreFoundation      	0x00007fff8e8661c4 __CFRunLoopServiceMachPort + 212
3   com.apple.CoreFoundation      	0x00007fff8e86568c __CFRunLoopRun + 1356
4   com.apple.CoreFoundation      	0x00007fff8e864ed8 CFRunLoopRunSpecific + 296
5   com.brave.brave.framework     	0x000000010dca7e0f 0x10dad0000 + 1932815
6   com.brave.brave.framework     	0x000000010dca7894 0x10dad0000 + 1931412
7   com.brave.brave.framework     	0x000000010dce3743 0x10dad0000 + 2176835
8   com.brave.brave.framework     	0x000000010dcd016d 0x10dad0000 + 2097517
9   com.brave.brave.framework     	0x000000010dd02aa8 0x10dad0000 + 2304680
10  com.brave.brave.framework     	0x000000010dcfec47 0x10dad0000 + 2288711
11  libsystem_pthread.dylib       	0x00007fff9734199d _pthread_body + 131
12  libsystem_pthread.dylib       	0x00007fff9734191a _pthread_start + 168
13  libsystem_pthread.dylib       	0x00007fff9733f351 thread_start + 13

Thread 7:: handle-watcher-thread
0   libsystem_kernel.dylib        	0x00007fff99f48db6 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0x00007fff97342728 _pthread_cond_wait + 767
2   com.brave.brave.framework     	0x000000010f33a738 0x10dad0000 + 25601848
3   com.brave.brave.framework     	0x000000010f31c90a 0x10dad0000 + 25479434
4   com.brave.brave.framework     	0x000000010f31c63c 0x10dad0000 + 25478716
5   com.brave.brave.framework     	0x000000010f2fed59 0x10dad0000 + 25357657
6   com.brave.brave.framework     	0x000000010f2feb5b 0x10dad0000 + 25357147
7   com.brave.brave.framework     	0x000000010f2feaa1 0x10dad0000 + 25356961
8   com.brave.brave.framework     	0x000000010dce3743 0x10dad0000 + 2176835
9   com.brave.brave.framework     	0x000000010dcd016d 0x10dad0000 + 2097517
10  com.brave.brave.framework     	0x000000010dd02aa8 0x10dad0000 + 2304680
11  com.brave.brave.framework     	0x000000010dcfec47 0x10dad0000 + 2288711
12  libsystem_pthread.dylib       	0x00007fff9734199d _pthread_body + 131
13  libsystem_pthread.dylib       	0x00007fff9734191a _pthread_start + 168
14  libsystem_pthread.dylib       	0x00007fff9733f351 thread_start + 13

Thread 8:: HTMLParserThread
0   libsystem_kernel.dylib        	0x00007fff99f42f72 mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x00007fff99f423b3 mach_msg + 55
2   com.apple.CoreFoundation      	0x00007fff8e8661c4 __CFRunLoopServiceMachPort + 212
3   com.apple.CoreFoundation      	0x00007fff8e86568c __CFRunLoopRun + 1356
4   com.apple.CoreFoundation      	0x00007fff8e864ed8 CFRunLoopRunSpecific + 296
5   com.brave.brave.framework     	0x000000010dca7e0f 0x10dad0000 + 1932815
6   com.brave.brave.framework     	0x000000010dca7894 0x10dad0000 + 1931412
7   com.brave.brave.framework     	0x000000010dce3743 0x10dad0000 + 2176835
8   com.brave.brave.framework     	0x000000010dcd016d 0x10dad0000 + 2097517
9   com.brave.brave.framework     	0x000000010dd02aa8 0x10dad0000 + 2304680
10  com.brave.brave.framework     	0x000000010dcfec47 0x10dad0000 + 2288711
11  libsystem_pthread.dylib       	0x00007fff9734199d _pthread_body + 131
12  libsystem_pthread.dylib       	0x00007fff9734191a _pthread_start + 168
13  libsystem_pthread.dylib       	0x00007fff9733f351 thread_start + 13

Thread 9:
0   libsystem_kernel.dylib        	0x00007fff99f49eca kevent + 10
1   com.brave.brave.framework     	0x000000010dbbad2c atom::NodeBindingsMac::PollEvents() + 140
2   com.brave.brave.framework     	0x000000010dbb99ff atom::NodeBindings::EmbedThreadRunner(void*) + 63
3   libnode.dylib                 	0x00000001121769b9 0x112023000 + 1391033
4   libsystem_pthread.dylib       	0x00007fff9734199d _pthread_body + 131
5   libsystem_pthread.dylib       	0x00007fff9734191a _pthread_start + 168
6   libsystem_pthread.dylib       	0x00007fff9733f351 thread_start + 13

Thread 10:: WorkerPool/32251
0   libsystem_kernel.dylib        	0x00007fff99f48db6 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0x00007fff9734275e _pthread_cond_wait + 821
2   com.brave.brave.framework     	0x000000010dcf201b 0x10dad0000 + 2236443
3   com.brave.brave.framework     	0x000000010dd0444e 0x10dad0000 + 2311246
4   com.brave.brave.framework     	0x000000010dd04832 0x10dad0000 + 2312242
5   com.brave.brave.framework     	0x000000010dcfec47 0x10dad0000 + 2288711
6   libsystem_pthread.dylib       	0x00007fff9734199d _pthread_body + 131
7   libsystem_pthread.dylib       	0x00007fff9734191a _pthread_start + 168
8   libsystem_pthread.dylib       	0x00007fff9733f351 thread_start + 13

Thread 11:: WorkerPool/25163
0   libsystem_kernel.dylib        	0x00007fff99f48db6 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0x00007fff9734275e _pthread_cond_wait + 821
2   com.brave.brave.framework     	0x000000010dcf201b 0x10dad0000 + 2236443
3   com.brave.brave.framework     	0x000000010dd0444e 0x10dad0000 + 2311246
4   com.brave.brave.framework     	0x000000010dd04832 0x10dad0000 + 2312242
5   com.brave.brave.framework     	0x000000010dcfec47 0x10dad0000 + 2288711
6   libsystem_pthread.dylib       	0x00007fff9734199d _pthread_body + 131
7   libsystem_pthread.dylib       	0x00007fff9734191a _pthread_start + 168
8   libsystem_pthread.dylib       	0x00007fff9733f351 thread_start + 13

Thread 12:: WorkerPool/33947
0   libsystem_kernel.dylib        	0x00007fff99f48db6 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0x00007fff9734275e _pthread_cond_wait + 821
2   com.brave.brave.framework     	0x000000010dcf201b 0x10dad0000 + 2236443
3   com.brave.brave.framework     	0x000000010dd0444e 0x10dad0000 + 2311246
4   com.brave.brave.framework     	0x000000010dd04832 0x10dad0000 + 2312242
5   com.brave.brave.framework     	0x000000010dcfec47 0x10dad0000 + 2288711
6   libsystem_pthread.dylib       	0x00007fff9734199d _pthread_body + 131
7   libsystem_pthread.dylib       	0x00007fff9734191a _pthread_start + 168
8   libsystem_pthread.dylib       	0x00007fff9733f351 thread_start + 13

Thread 13:: WorkerPool/7899
0   libsystem_kernel.dylib        	0x00007fff99f48db6 __psynch_cvwait + 10
1   libsystem_pthread.dylib       	0x00007fff9734275e _pthread_cond_wait + 821
2   com.brave.brave.framework     	0x000000010dcf201b 0x10dad0000 + 2236443
3   com.brave.brave.framework     	0x000000010dd0444e 0x10dad0000 + 2311246
4   com.brave.brave.framework     	0x000000010dd04832 0x10dad0000 + 2312242
5   com.brave.brave.framework     	0x000000010dcfec47 0x10dad0000 + 2288711
6   libsystem_pthread.dylib       	0x00007fff9734199d _pthread_body + 131
7   libsystem_pthread.dylib       	0x00007fff9734191a _pthread_start + 168
8   libsystem_pthread.dylib       	0x00007fff9733f351 thread_start + 13

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000501  rbx: 0x0000000000000000  rcx: 0x00007f966a60da01  rdx: 0x0000260a085acc79
  rdi: 0x00007fff783f7000  rsi: 0x00007f966b810c00  rbp: 0x00007fff52135a00  rsp: 0x00007fff52135a00
   r8: 0x00007f966a628230   r9: 0x0000000000000015  r10: 0x0000000000000150  r11: 0x0000287e430ed2b9
  r12: 0x0000000000000000  r13: 0x0000260a085041e9  r14: 0x00007f966b810c00  r15: 0x00007f966a82b838
  rip: 0x000000010fed773f  rfl: 0x0000000000010202  cr2: 0x0000700000081000
  
Logical CPU:     0
Error Code:      0x02000131
Trap Number:     133
```

**Comment by bridiver** [comment]
are you running in dev or did you download the app? If you are running in dev, can you delete node_modules and ~/.electron and try again? Any specific steps that reproduce this?

**Comment by fiery-** [comment]
I downloaded it from brave.com. About Box shows:
```
Brave: 0.9.6
Electron: 0.37.8
libchromiumcontent: 50.0.2661.102
V8: 5.0.71.48
Node.js: 5.10.0
Update channel: dev
```
Currently I can reproduce it, like this:
1. Restart the machine OS X 10.11.4
2. Start Brave and wait a little to connect to amazonaws.com servers (I allow it in Little Snitch).
3. Quit Brave - crash occurs at this point, if not start Brave again and after a few seconds Brave Helper crashes.

Start-page is set to Homepage, which is empty. At first I was not able to reproduce the problem, but after starting Vivaldi v1.1 (I wrote about:flags and closed Vivaldi), then after several restarts the problem appears each time as described above.

It seems that there are other similar crashes in Electron: https://github.com/electron/electron/issues/4299
